corehq/apps/app_manager/models.py

```diff
@@ -1519,18 +1520,48 @@ class MappingItem(DocumentSchema):
     value = DictProperty()
 
     @property
+    def treat_as_expression(self):
+        """
+        Returns if whether the key can be treated as a valid expression that can be included in
+        condition-predicate of an if-clause for e.g. if(<expression>, value, ...)
+        """
+        special_chars = '{}()[]=<>."\'/'
+        return any(special_char in self.key for special_char in special_chars)
+
+    @property
     def key_as_variable(self):
         """
         Return an xml variable name to represent this key.
-        If the key has no spaces, return the key with "k" prepended.
-        If the key does contain spaces, return a hash of the key with "h" prepended.
+
+        If the key contains spaces or a condition-predicate of an if-clause,
+        return a hash of the key with "h" prepended.
+        If not, return the key with "k" prepended.
+
         The prepended characters prevent the variable name from starting with a
         numeral, which is illegal.
         """
-        if " " not in self.key:
+        if ' ' in self.key or self.treat_as_expression:
+            return 'h{hash}'.format(hash=hashlib.md5(self.key).hexdigest()[:8])
+        else:
             return 'k{key}'.format(key=self.key)
+
+    def key_as_condition(self, property):
+        if self.treat_as_expression:
+            condition = dot_interpolate(self.key, property)
+            return u"{condition}".format(condition=condition)
         else:
-            return 'h{hash}'.format(hash=hashlib.md5(self.key).hexdigest()[:8])
+            return u"{property} = '{key}'".format(
+                property=property,
+                key=self.key
+            )
+
+    def key_as_xpath_variable(self, index, type):
```

Comment:
slight preference for using a boolean like `is_sort` rather than a `type` variable if there are only two options (easier to know what's going on without having to read the function)